### PR TITLE
Prepare tokio-util 0.6.10

### PR DESCRIPTION
**Pull request by Darksonn**
# 0.6.10 (May 14, 2021)

This is a backport for a memory leak in `CancellationToken`. ([#4652])

[#4652]: https://github.com/tokio-rs/tokio/pull/4652

### CODE REVIEW

**Comment by hawkw** [comment]
looks like CI is failing because... the version of Tokio on this commit produces a warning on the latest compiler: https://github.com/tokio-rs/tokio/runs/6435358613?check_suite_focus=true 🙃 

i'm not really sure what's the best way to handle this — it shouldn't block the `tokio-util` release, but it's annoying that we'll have to update a past `tokio` to get CI to pass...

**Comment by hawkw** [comment]
it would be nice if there was a way to disable `-D warnings` on CI but *only for backports*, but actually implementing this probably requires a bunch of annoying GitHub Actions wrangling...

**Comment by Darksonn** [comment]
Since this wont be merged with a squash, I will need a force push to squash these CI fix attempts before merging, but for now I'll just push to experiment.

**Review by hawkw — APPROVED** [review]
lgtm, just changing the CI configs for the backport seems like a fine solution tbh

**Comment by hawkw** [comment]
ugh, i have no memory of what the CircleCI build was for...https://app.circleci.com/pipelines/github/tokio-rs/tokio/657/workflows/93a890e9-15f5-4ba8-a3ae-3e604bb107bd/jobs/646

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: 2.1        
+jobs:
+  test-arm:
+    machine:
+      image: ubuntu-2004:202101-01
+    resource_class: arm.medium
+    environment:
+      # Change to pin rust versino
```
Comment (renancloudwalk) [inline .circleci/config.yml:8]: Typo

File: tokio-util/src/sync/cancellation_token.rs [hunk 2]
```diff
@@ -203,78 +144,42 @@ impl CancellationToken {
...
     /// derived from it.
     ///
     /// This will wake up all tasks which are waiting for cancellation.
+    ///
+    /// Be aware that cancellation is not an atomic operation. It is possible
+    /// for another thread running in parallel with a call to `cancel` to first
+    /// receive `true` from `is_cancelled` on one child node, and then receive
+    /// `false` from `is_cancelled` on another child node. However, once the
+    /// call to `cancel` returns, all child nodes have been fully cancelled.
     pub fn cancel(&self) {
```
Comment (renancloudwalk) [inline tokio-util/src/sync/cancellation_token.rs:263]: Shouldn't it return a result, just in case some concurrent process tries to cancel it?

**Review by renancloudwalk — APPROVED** [review]
Nice improvement towards safety 👍

File: .circleci/config.yml (hunk 1 above)
Comment (Darksonn) [reply .circleci/config.yml:8]: This was simply copied from master. Feel free to open a PR that fixes it on master.

File: tokio-util/src/sync/cancellation_token.rs (hunk 2 above)
Comment (Darksonn) [reply tokio-util/src/sync/cancellation_token.rs:263]: No, the token doesn't tell you if it was already cancelled. Regardless, this PR is not the place to discuss it.

**Pull request merged**